### ISSUE TITLE
Questions about calculating gradients and updating weights

### ISSUE THREAD

**Issue by kwonmha** [erwtokritos/collaborativefiltering-rbm#1]
I'm not sure if this is the right place to write questions but couldn't find anywhere to do.
So I write here.
This code is really helpful but I can't understand the process of calculating gradients and updating weights.
Your rbm updates weights for every different users.
And your hidbiasinc, bisbiasesInc, Wijk_inc continually changing as your rbm processing every users in one epoch step.
So as a result, the weights will be updated by those *inc variables that have been changed by that moment.
In other words, the latter the weights are updated, the more accumulated value will be used, though they are used in same epoch.
(to summarize my question)
1. I think the inc variables should be initialized to zero everytime rbm processing for different users.
2. Or weights should be updated once in every epoch, not every users, while they remain unchanged during one epoch.

Thx
